lib/matplotlib/backends/backend_ps.py

```diff
@@ -1188,51 +1186,39 @@ def write(self, *args, **kwargs):
-            with open(tmpfile, 'w', encoding='latin-1') as fh:
-                # write the Encapsulated PostScript headers
-                print("%!PS-Adobe-3.0 EPSF-3.0", file=fh)
-                if title:
-                    print("%%Title: "+title, file=fh)
-                # get source date from SOURCE_DATE_EPOCH, if set
-                # See https://reproducible-builds.org/specs/source-date-epoch/
-                source_date_epoch = os.getenv("SOURCE_DATE_EPOCH")
-                if source_date_epoch:
-                    source_date = datetime.datetime.utcfromtimestamp(
-                        int(source_date_epoch)).strftime(
-                            "%a %b %d %H:%M:%S %Y")
-                else:
-                    source_date = time.ctime()
-                print(
-                    "%%Creator: {creator_str}\n"
-                    "%%CreationDate: {source_date}\n"
-                    "%%BoundingBox: {bbox[0]} {bbox[1]} {bbox[2]} {bbox[3]}\n"
-                    "%%EndComments\n"
-                    .format(creator_str=creator_str, source_date=source_date,
-                            bbox=bbox),
-                    end="", file=fh)
-
-                print("%%BeginProlog\n"
-                      "/mpldict {len_psDefs} dict def\n"
-                      "mpldict begin\n"
-                      "{psDefs}\n"
-                      "end\n"
-                      "%%EndProlog\n"
-                      .format(len_psDefs=len(psDefs),
-                              psDefs="\n".join(psDefs)),
-                      end="", file=fh)
-
-                print("mpldict begin", file=fh)
-                print("%s translate" % _nums_to_str(xo, yo), file=fh)
-                print("%s clipbox" % _nums_to_str(width*72, height*72, 0, 0),
-                      file=fh)
-
-                # write the figure
-                print(self._pswriter.getvalue(), file=fh)
-
-                # write the trailer
-                print("end", file=fh)
-                print("showpage", file=fh)
-                fh.flush()
+            # get source date from SOURCE_DATE_EPOCH, if set
+            # See https://reproducible-builds.org/specs/source-date-epoch/
+            source_date_epoch = os.getenv("SOURCE_DATE_EPOCH")
+            if source_date_epoch:
+                source_date = datetime.datetime.utcfromtimestamp(
+                    int(source_date_epoch)).strftime("%a %b %d %H:%M:%S %Y")
+            else:
+                source_date = time.ctime()
+            pathlib.Path(tmpfile).write_text(
+                # Because Title can be an empty line ("Title: <empty>" is
+                # different from no title at all) and empty lines also
+                # terminate the header, it must come just before %%EndComments.
+                f"""\
+%!PS-Adobe-3.0 EPSF-3.0
+%%Creator: {creator_str}
+%%CreationDate: {source_date}
+%%BoundingBox: {bbox[0]} {bbox[1]} {bbox[2]} {bbox[3]}
+{f"%%Title: {title}" if title else ""}
```

Comment:
Is it ok that this may become a blank line?
Also the order is changed. Is that intentional?

Reply:
Yes per the comment above, but I figured out a better way...